docs/content/manual/manual.yml

```diff
@@ -2098,8 +2098,14 @@ sections:
           * `@sh`:
 
             The input is escaped suitable for use in a command-line
-            for a POSIX shell. If the input is an array, the output
-            will be a series of space-separated strings.
+            for a POSIX shell using `eval`. If the input is an array,
+            the output will be a series of space-separated strings.
+            If the input is an object, the output will be a series of
+            space-separated variable assignments for all the keys in
+            the object that are valid shell variable names (other keys
+            will be ignored).
+
+            E.g., `eval $(jq -r '@sh' f.json)`
```

Comment:
I don't understand what this example is supposed to show, why would you run `@sh` to quote everything into a single quoted shell word, and use `eval`? It would be the exact same thing as running `"$(jq -r . f.json)"`. But, anyway, it is still wrong.
You need to quote the expansion, and arbitrary strings passed to eval need to be prefixed by a space to avoid making the shell interpret a string that  starts with `-` as an option to `eval`: `eval " $(foo)"`

Most of the times I use `@sh` is not with `eval`, but with `| sh`:
```sh
jq -r '.things[] | @sh "cmd -o \(.foo) -- \(.majigs)"' | sh
```

If we want to put an example, it should at least make sense.

Reply:
Oh, this is supposed to be an example for the new `@sh` behaviour for objects that creates a string of assignments.
e.g. `jq -n '{foo:1, bar:2} | @sh'` => `foo='1' bar='2'`.
You still need to use proper quoting to make this actually work, but at least now I get it. ^^

```
eval " $(jq -r '@sh' file.json)"
```